server/lib/cache-api-user-has-run-as.ts

```diff
@@ -51,28 +51,46 @@ export const APIUserAllowRunAs = {
         {},
         { apiHostID: apiId }
       );
-      const statusUserAllowRunAs = response.data.data.affected_items[0].allow_run_as ? API_USER_STATUS_RUN_AS.ENABLED : API_USER_STATUS_RUN_AS.NOT_ALLOWED;
+      const statusUserAllowRunAs = response.data.data.affected_items[0].allow_run_as ? API_USER_STATUS_RUN_AS.ENABLED : API_USER_STATUS_RUN_AS.INTERFACE_DISABLED;
 
       // Cache the run_as for the API user
       CacheInMemoryAPIUserAllowRunAs.set(apiId, api.username, statusUserAllowRunAs);
       return statusUserAllowRunAs;
     }catch(error){
       log('APIUserAllowRunAs:check', error.message || error);
-      return API_USER_STATUS_RUN_AS.DISABLED;
+      return API_USER_STATUS_RUN_AS.ALL_DISABLED;
     }
   },
   async canUse(apiId: string): Promise<number | never>{
     const ApiUserCanUseStatus = await APIUserAllowRunAs.check(apiId);
-    if(ApiUserCanUseStatus === API_USER_STATUS_RUN_AS.NOT_ALLOWED){
+    if(ApiUserCanUseStatus === API_USER_STATUS_RUN_AS.HOST_DISABLED){
       const api = await manageHosts.getHostById(apiId);
       throw new Error(`API with host ID [${apiId}] misconfigured. The Wazuh API user [${api.username}] is not allowed to use [run_as]. Give it permissions or set [run_as] host setting with [false] value.`);
-    };
+    }
     return ApiUserCanUseStatus;
   }
 };
 
+/**
+ * @example
+ *   HOST = set in wazuh.yml config
+ *   INTERFACE = set in user interface
+ *
+ * ALL_DISABLED
+ *   binary 00 = decimal 0 ---> INTERFACE 0 y HOST 0
+ * 
+ * INTERFACE_DISABLED
+ *   binary 01 = decimal 1 ---> INTERFACE 0 y HOST 1
+ * 
+ * HOST_DISABLED
+ *   binary 10 = decimal 2 ---> INTERFACE 1 y HOST 0
+ * 
+ * ENABLED
+ *   binary 11 = decimal 3 ---> INTERFACE 1 y HOST 1
+ */
 export enum API_USER_STATUS_RUN_AS{
-  NOT_ALLOWED = -1, // Wazuh API user configured with run_as in wazuh.yml but it has not run_as in Wazuh API
-  DISABLED = 0, // Wazuh API user configured with run_as=false or undefined
-  ENABLED = 1 // Wazuh API user configured with run_as=true and allow run_as
+  ALL_DISABLED = 0, // Wazuh HOST and INTERFACE API user configured with run_as=false or undefined
+  INTERFACE_DISABLED = 1, // Wazuh HOST API user configured with run_as = TRUE in wazuh.yml but it has not run_as in Wazuh API
```

Comment:
I think the `INTERFACE_DISABLED` variable could be a little more clear with his real meaning. `Interface` entity sounds a little rare to me.

It could have 2 entities: `user` and `host` refering to `run_as` configuration:
- User
  - allowed
  - not allowed
- Host (configuration in `wazuh.yml`)
  - enabled
  - disabled

Reply:
Hi Desvelao,
sounds about right, so I changed INTERFACE_DISABLED to USER_NOT_ALLOWED